### PR TITLE
Add Stories

### PR DESCRIPTION
**Pull request by HarishTeens**

### CODE REVIEW

**Comment by roshankshaw** [comment]
Roles and permissions need to be done

**Comment by HarishTeens** [comment]
permissions for addStory and getCurrentStory has to be set. Will do it

**Comment by DesignrKnight** [comment]
Will be done as in, will be done in separate PR I suppose?

File: sources/datasources/stories.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    initialize(config) {}
+    getStories(args) {
+		return Stories.find(args);
+	}
+	async getCurrentStories() {
+        let activeStories=await currentStories.find();
+        return activeStories;        
+	}
+	async addStory(story) {
+		let retPromise = {};
```
Comment (DesignrKnight) [inline sources/datasources/stories.js:26]: retPromise?
meaning?

File: sources/datasources/stories.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            createdStory.author = foundAuthor._id;
+
+            //add to current Stories
+            await currentStories.create({
+                storyAsset:story.asset,                
+                authorLogo: foundAuthor.logo,
+                authorName: foundAuthor.clubName,
+                storyID: createdStory._id
+            })
```
Comment (DesignrKnight) [inline sources/datasources/stories.js:51]: Do we need an asset type in the model?
How to know if it is image or video down the line without trying to render the asset

File: sources/models/currentStory.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/** @format */
+
+var mongoose = require('mongoose');
+var currentStorySchema = new mongoose.Schema({
+	storyAsset:String,
+	authorName: String, 
+	authorLogo: String,
```
Comment (DesignrKnight) [inline sources/models/currentStory.js:7]: Make these as required: true

File: sources/models/story.js [hunk 4]
```diff
@@ -1,16 +1,21 @@
 /** @format */
 
 var mongoose = require('mongoose');
-var eventSchema = new mongoose.Schema({
-	storyId: Number,
+var storySchema = new mongoose.Schema({
 	author: {
 		type: mongoose.Schema.Types.ObjectId,
 		ref: 'Club',
 	},
```
Comment (DesignrKnight) [inline sources/models/story.js:8]: even this as required:true

**Review by DesignrKnight — COMMENTED** [review]
I am not completely understanding the logic because I was not the developer😂
Also, run prettier, code looks weirdly formatted

File: sources/models/currentStory.js (hunk 3 above)
Comment (roshankshaw) [reply sources/models/currentStory.js:7]: We will work on input validation on all the schemas

File: sources/datasources/stories.js (hunk 1 above)
Comment (roshankshaw) [reply sources/datasources/stories.js:26]: We need to return a promise to the resolver when this function is called

File: sources/datasources/stories.js (hunk 2 above)
Comment (roshankshaw) [reply sources/datasources/stories.js:51]: asset will be a cloudinary link

**Comment by roshankshaw** [comment]
> I am not completely understanding the logic because I was not the developer😂
> Also, run prettier, code looks weirdly formatted

Earlier it was brute force, then we came up with Jugaad, then Harish found (Jugaad)^2.😂

File: sources/datasources/stories.js (hunk 2 above)
Comment (DesignrKnight) [reply sources/datasources/stories.js:51]: yeah, but how will client know from link if it is video or image

File: sources/datasources/stories.js (hunk 1 above)
Comment (DesignrKnight) [reply sources/datasources/stories.js:26]: the variable name I am asking, 'retPromise'

Comment (DesignrKnight) [reply sources/datasources/stories.js:26]: short for returnPromise?

File: sources/datasources/stories.js (hunk 2 above)
Comment (roshankshaw) [reply sources/datasources/stories.js:51]: @Thesmader is working on that.

File: sources/datasources/stories.js (hunk 1 above)
Comment (roshankshaw) [reply sources/datasources/stories.js:26]: Yes it is short form for return Promise

**Review by roshankshaw — APPROVED** [review]

File: sources/datasources/stories.js (hunk 1 above)
Comment (DesignrKnight) [reply sources/datasources/stories.js:26]: put better name please. we will graduate soon and we want people to understand code. retPromise is weird

**Review by roshankshaw — APPROVED** [review]
Changes concerning StoryAPI architecture are alright, The Input Validation, Roles and Permissions and Cloudinary integration is to be checked

**Comment by DesignrKnight** [comment]
Shall I merge?
Let us merge, if needed, we can patch fix later.

**Comment by Thesmader** [comment]
Writing a pull request description goes a long way in making it easier for the reviewer to understand what the PR is, where to look at, what to look at and what to expect 🌚🌝.

File: sources/datasources/stories.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+		retPromise = await createdStory.save();
+		return retPromise;
+    }
+    
+    async deleteStory(story){
+        await Stories.deleteOne({ "_id" : story.id } )
+        await currentStories.deleteOne({ "storyID" : story.id } )
+        return {success:true}
+    }
```
Comment (DesignrKnight) [inline sources/datasources/stories.js:77]: delete story only if story is active, else do not provide option

**Comment by DesignrKnight** [comment]
@Thesmader confirm soon if we need asset type field along with asset link? Or will asset link be sufficient on front-end?

**Comment by Thesmader** [comment]
The type parameter should be available for query in the API.

File: sources/models/story.js [hunk 6]
```diff
@@ -1,16 +1,21 @@
...
-	posted_at: String,
-	picture: String,
+	event: {
+		type: mongoose.Schema.Types.ObjectId,
+		ref: 'Event',
+	},
```
[the remark this answers is not in the document]
Comment (DesignrKnight) [reply sources/models/story.js:12]: because we need to have swipe up to event

File: sources/datasources/stories.js (hunk 5 above)
Comment (HarishTeens) [reply sources/datasources/stories.js:77]: so basically archive story?

**Comment by DesignrKnight** [comment]
Hey @HarishTeens @roshankshaw 
If this is ready to merge, do merge
Edits and cleanup can be in separate PR also. Your call

**Comment by HarishTeens** [comment]
> Hey @HarishTeens @roshankshaw 
> If this is ready to merge, do merge
> Edits and cleanup can be in separate PR also. Your call

Yepa yeap, it's ready to merge. Have created a new branch for cleanup and docs. 💪

**Pull request merged**